internal/relationaldb/account.go

```diff
@@ -0,0 +1,194 @@
+* limitations under the License.
+ */
+
+package relationaldb
+
+import (
+	"context"
+	"fmt"
+	"github.com/coinbase-samples/ib-ledger-go/internal/utils"
+
+	"github.com/coinbase-samples/ib-ledger-go/internal/model"
+	"github.com/google/uuid"
+)
+
+const (
+	insertAccountSql = `
+        INSERT INTO account (id, qldb_id, user_id, currency) 
+        VALUES ($1, $2, $3, $4) 
+        ON CONFLICT (id)
+        DO NOTHING`
+
+	selectAccountById = `SELECT id FROM account WHERE id = $1`
+
+	insertAccountBalanceSql = `
+        INSERT INTO account_balance (account_id, balance, hold, available, created_at, idem)
+        VALUES ($1, $2, $3, $4, $5, $6)
+        ON CONFLICT ON CONSTRAINT balance_change
+        DO NOTHING`
+
+	getAllAccountsAndBalancesSql = `
+        SELECT acc.id, acc.currency, ab.balance, ab.hold, ab.available, ab.created_at
+        FROM (select id, currency FROM account WHERE user_id = $1) acc
+        INNER JOIN
+        (SELECT account_id, MAX(count) as max
+        FROM account_balance
+        WHERE account_id IN (select id
+        FROM account
+        WHERE user_id = $1)
+        GROUP BY account_id) recent_balance
+        ON acc.id = recent_balance.account_id
+        INNER JOIN
+        account_balance ab
+        ON recent_balance.account_id = ab.account_id and recent_balance.max = ab.count
+        GROUP BY ab.count, acc.id, acc.currency, ab.balance, ab.hold, ab.available, ab.created_at
+        HAVING recent_balance.count = acc.count;`
+
+	getAccountByIdSql = `SELECT * FROM account WHERE id = $1`
+)
+
+func insertAccount(ctx context.Context, account *model.Account) error {
+	if err := Repo.Insert(
+		ctx,
+		insertAccountSql,
+		account.Id.String(),
+		account.QldbId,
+		account.UserId.String(),
+		account.Currency,
+	); err != nil {
+		return fmt.Errorf(
+			"failed to insert account - id: %s - %w",
+			account.Id,
+			err,
+		)
+	}
+	return nil
+}
+
+func InsertAccountBalance(
+	ctx context.Context,
+	account *model.QldbAccount,
+) error {
+	id := account.AccountUUID
+	userId := account.UserId
+	currency := account.Currency
+	var accountId []*uuid.UUID
+	if err := Repo.Query(
+		ctx,
+		&accountId,
+		selectAccountById,
+		id,
+	); err != nil {
+		return fmt.Errorf(
+			"failed to retrieve account "+
+				"- accountId: %s - userId: %s - currency: %s - %w",
+			id,
+			userId,
+			currency,
+			err,
+		)
+	}
+
+	// If first attempt doesn't find an accountId, create the new account
+	if len(accountId) == 0 {
+		if err := Repo.Insert(
+			ctx,
+			insertAccountSql,
+			id,
+			account.Id,
+			userId,
+			currency,
+		); err != nil {
+			return fmt.Errorf(
+				"failed to create account "+
+					"- accountId: %s - userId: %s - currency: %s - %w",
+				id,
+				userId,
+				currency,
+				err,
+			)
+		}
+	}
+
+	balance, _ := account.Balance.CoEx()
+	hold, _ := account.Hold.CoEx()
+	available, _ := account.Available.CoEx()
+	idem := utils.GenerateIdemString(
```

Comment:
If you have multiline blocks like this, you might as well add a ',' on the last line, so that the block indention is consistent

This is all over your code:

E.g., 

idem := utils.GenerateIdemString(
                 id,
		balance.String(),
		hold.String(),
		available.String(),
		account.UpdatedAt.String(),
)